src/ledger.ts

```diff
@@ -0,0 +1,164 @@
+// LedgerSystem class
+import BigNumber from "bignumber.js";
+import * as dataConnectors from "./data-connectors";
+import { FIFOQueue } from "./queue";
+import { sleep, groupBy } from "./utils";
+import { IPostingEntry, BaseDataConnector, IOperation, OperationStatus, OperationType} from "./base-data-connector";
+export interface IPostingEntryRequest {
+    bookId: string,
+    assetId: string,
+    value: string,
+}
+export interface IOperationRequest {
+    type: OperationType,
+    memo: string,
+    entries: IPostingEntryRequest[],
+}
+export interface ITransferRequest {
+    fromBookId: string,
+    toBookId: string,
+    assetId: string,
+    value: string,
+    memo: string,
+}
+export interface IBookRequest {
+    name: string,
+    metadata?: {},
+    restrictions?: {
+        minBalance?: string;
+    }
+}
+export interface IBookBalances {
+    [assetId: string]: string,
+}
+export class LedgerSystem {
```

Comment:
You need to expose this additional functionality - 

'Get operations on book' -> paginated way to get latest completed operations on book.

Callers need it for several use cases.